<header>
    <div style="text-align: center;">
        <h1> MSc Metropolitan Analysis Design Engineering </h1>
        <h1> Metropolitan Data 1 </h1>
        <h3> Stef Kraaij, Eirini Terezaki, Marco Russo </h3>
        <h4> <i>Assignment 3</i> </h4>
    </div>
</header>

For the forth assignment, we created a swimming path using some specific points and coordinates on the map. We created a html file, that can be opened with a web browser. That file contains a map centred around Amsterdam. With the list of the specific coordinates we choose, we created a polyline object that connects them. To reach that result we used this code: 

In [2]:
import folium
import os

map_amsterdam = folium.Map(location=[52.3667, 4.8945], zoom_start=13)

coordinates = [(52.348285, 4.911252), (52.358752, 4.904938), (52.364527, 4.901816),
(52.365808, 4.908862), (52.363347, 4.916383), (52.365858, 4.921591),
(52.369648, 4.910720), (52.371231, 4.913101), (52.373051, 4.914184),
(52.373275, 4.916235)]

path = folium.PolyLine(locations=coordinates, color='blue', weight=2)

path.add_to(map_amsterdam)

folder_path = "C:\\Users\\eterezakii\\Desktop\\AMS Stuff"

os.makedirs(folder_path, exist_ok=True)

file_path = os.path.join(folder_path, 'amsterdam_map.html')
map_amsterdam.save(file_path)

In order to create the file, we had to define the path of the folder in the computer, in which the file will be placed. For the code to work with another user, the path must be changed to a new one.

**1. Find the centre of the nodes of the swimming route.**

In order to calculate the central point of the swimming path, we created this code:

In [1]:

nodes = [(52.348285, 4.911252), (52.358752, 4.904938),(52.364527, 4.901816),(52.365808, 4.908862),(52.363347, 4.916383),(52.365858, 4.921591),(52.369648, 4.910720),(52.371231, 4.913101),(52.373051, 4.914184),(52.373275, 4.916235)]

total_lat = 0
total_lon = 0


for node in nodes:
    total_lat += node[0]
    total_lon += node[1]


center_lat = total_lat / len(nodes)
center_lon = total_lon / len(nodes)


print(f"Center coordinates: ({center_lat}, {center_lon})")

Center coordinates: (52.365378199999995, 4.911908199999999)


The result we get from running this script is these coordinates: (52.365378199999995, 4.911908199999999)
So this is the point that is in the middle of our swimming route.

**2. Use the centre to find a suitable spot for the Event Headquarters.**

In the previous question we found the center of our swimming route. The center is around the Amsterdam Business school. That school is also along the route next to the water, so it has direct connection with the swimming route. Because it is Amsterdam Business School, we believe that it is the perfect place to use as the event headquarters as it will probably have facilities to accomodate the headquarters of the event. 

**3. Find the closest bus and tram stops at the start and finish of the swimming route. How
many people can be transported within an hour.**

The closest tram station from our event headquarters is the tram station called ‘Artis’. This station has tramlines 14, 4  and bus lines N87 and N89. 

Tram 4 is four times an hour either direction.
Tram 14 is four times an hour either direction.
Given that Bus N87 and N89 are nightbuses, we don't take them into accont for the assignment.

So, according to GVB the most common tram is the G13 or G14. These trams have a capacity of 125 or 150 respectively. So let's take the average which is 138.

So there are 8 trams an hour, with an average capacity of 138. 


In [4]:
Average_Capacity = 138
Trams = 8

Flowrate_hour = Trams * Average_Capacity

print(Flowrate_hour)

1104


The amount of people that can be transported within an hour through these trams is 1104 people.

**4. Can you find which bus and tram lines these are, and can you find their routes?**

We made a map just like in the first question. More specifically,we updated the code which we used for the mapping of our path. Now both the swimming route and the tram route are visible on the map in the html file. The difference now is that along with the swimming path, the route of Tram 14 is also visible.

In [6]:
import folium
import os

map_amsterdam = folium.Map(location=[52.3667, 4.8945], zoom_start=13)

coordinates1 = [(52.348285, 4.911252), (52.358752, 4.904938), (52.364527, 4.901816),
(52.365808, 4.908862), (52.363347, 4.916383), (52.365858, 4.921591),
(52.369648, 4.910720), (52.371231, 4.913101), (52.373051, 4.914184),
(52.373275, 4.916235)]

coordinates2 = [(52.377617, 4.901658), (52.377946, 4.899388), (52.373118, 4.893331),
(52.369288, 4.892183), (52.367063, 4.893618), (52.366283, 4.896869),
(52.366252, 4.900320), (52.367188, 4.902159), (52.368037, 4.905431),
(52.366699, 4.911111), (52.363121, 4.920102), (52.365912, 4.927030),
(52.366030, 4.932353), (52.363754, 4.939550), (52.361622, 4.939744),
(52.362623, 4.946752), (52.365135, 4.947992), (52.365238, 4.951205)]

path1 = folium.PolyLine(locations=coordinates1, color='blue', weight=2)
path2 = folium.PolyLine(locations=coordinates2, color='red', weight=2)

path1.add_to(map_amsterdam)
path2.add_to(map_amsterdam)

folder_path = "C:\\Users\\eterezakii\\Desktop\\AMS Stuff"

os.makedirs(folder_path, exist_ok=True)

file_path = os.path.join(folder_path, 'amsterdam_map.html')
map_amsterdam.save(file_path)


**5. Calculate the centrality of the start, finish, and centre node of the route. Which
centrality calculation makes the most sense.**

We defined centrality as the relation of the starting point, the ending point of the path and the central point we calculated earlier with the street network of Amsterdam. To be more specific, we calculated the centrality based on the street system of the area around our path and how easy it is to get to these points. If the point of research  gets 0 that it has no centrality and if it has 1 it has 100% centrality.

In [1]:
import osmnx as ox
import networkx as nx
import folium

start_point = (52.348285, 4.911252)
end_point = (52.373275, 4.916235)
new_point = (52.365378199999995, 4.911908199999999)

graph = ox.graph.graph_from_point(start_point, dist=2000, network_type='all')

centrality = nx.closeness_centrality(graph)

start_node = ox.distance.nearest_nodes(graph, start_point[0], start_point[1])
end_node = ox.distance.nearest_nodes(graph, end_point[0], end_point[1])
new_node = ox.distance.nearest_nodes(graph, new_point[0], new_point[1])


start_centrality = centrality[start_node]
end_centrality = centrality[end_node]
new_centrality = centrality[new_node]

map_center = folium.Map(location=start_point, zoom_start=13)


folium.Marker(start_point, icon=folium.Icon(color='blue')).add_to(map_center)
folium.Marker(end_point, icon=folium.Icon(color='red')).add_to(map_center)
folium.Marker(new_point, icon=folium.Icon(color='green')).add_to(map_center)


folium.Marker(start_point, icon=folium.Icon(color='blue'), popup=f'Starting Centrality: {start_centrality}').add_to(map_center)
folium.Marker(end_point, icon=folium.Icon(color='red'), popup=f'Ending Centrality: {end_centrality}').add_to(map_center)
folium.Marker(new_point, icon=folium.Icon(color='green'), popup=f'New Point Centrality: {new_centrality}').add_to(map_center)

file_path = "C:\\Users\\eterezakii\\Desktop\\AMS Stuff\\centrality_map.html"
map_center.save(file_path)


So, the result is a file of a map that when you click on the three points of interest you get the centrality of them: 

 **6. Find all cafes, restaurants near the finish line. Walking time smaller than 10 minutes.**

The number of horeca we've found within 1km distance from the end point is 115


In [2]:
import pandas as pd
from geopy.distance import geodesic

pd.set_option('display.max_columns', None)
pd.set_option('display.expand_frame_repr', False)
pd.set_option('display.max_rows', None)

pd.set_option('display.max_rows', None)

df = pd.read_csv("C:\\Users\\Stef_\\OneDrive\\Documenten\\MADE\\Metropolitan Data\\Github repository Data 1\\Assignment 4 Transport\\FUNCTIEKAART.csv", delimiter=';')


horeca_df = df[df['Functie'] == 'Horeca']

end_location = (52.373275, 4.916235)


def extract_coordinates(row):
    
    coordinates_str = row['WKT_LAT_LNG'].split('(')[1].rstrip(')').split()
    return (float(coordinates_str[0]), float(coordinates_str[1]))


horeca_df = horeca_df.copy()

horeca_df['Coordinates'] = horeca_df.apply(extract_coordinates, axis=1)
horeca_df['Distance'] = horeca_df['Coordinates'].apply(lambda coords: geodesic(end_location, coords).meters)


close_horeca = horeca_df[horeca_df['Distance'] <= 1000]

num_horeca_found = len(close_horeca)

print(f"The number of horeca we've found within 1km distance from the end point is {num_horeca_found}")


print("Here is the list of the found horecas:")

print(close_horeca[['OBJECTNUMMER', 'Adres', 'Gebruiksdoel', 'Objectsoort']])

FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\Stef_\\OneDrive\\Documenten\\MADE\\Metropolitan Data\\Github repository Data 1\\Assignment 4 Transport\\FUNCTIEKAART.csv'

With this code we were able to print the coordinates the points of interest and then use them to create a map with them so we can visually see where they are in the map.

In [3]:
import folium


point_coordinates = [
    (52.3739837, 4.9019093),
    (52.3738505, 4.9022631),
    (52.3736277, 4.9027502),
    (52.3739038, 4.9021158),
    (52.3738862, 4.9022041),
    (52.3675904, 4.9258969),
    (52.36794, 4.9256583),
    (52.3698595, 4.9293091),
    (52.3706896, 4.9301375),
    (52.3686264, 4.9265174),
    (52.369371, 4.9284919),
    (52.3694525, 4.9286525),
    (52.3696968, 4.929076),
    (52.3777809, 4.9051722),
    (52.3777809, 4.9051722),
    (52.376288, 4.9116205),
    (52.3688658, 4.9142548),
    (52.3681012, 4.9163771),
    (52.3696563, 4.9119117),
    (52.3754778, 4.9024069),
    (52.3754598, 4.9024072),
    (52.3758379, 4.9025791),
    (52.3740819, 4.9017172),
    (52.3739738, 4.9016744),
    (52.3760709, 4.9068354),
    (52.3701717, 4.9126844),
    (52.3714067, 4.9273403),
    (52.3766592, 4.9213824),
    (52.3771131, 4.9202322),
    (52.3702783, 4.9123895),
    (52.3701804, 4.9126109),
    (52.3700338, 4.9119224),
    (52.3701155, 4.9121123),
    (52.3702151, 4.9123169),
    (52.3708874, 4.9163622),
    (52.372411, 4.9019116),
    (52.3723484, 4.9020153),
    (52.3720707, 4.9022237),
    (52.3717866, 4.9030933),
    (52.3714876, 4.9024797),
    (52.3684723, 4.9058016),
    (52.3674136, 4.9062535),
    (52.3750047, 4.9304309),
    (52.3678938, 4.9253505),
    (52.3678943, 4.9254825),
    (52.3692605, 4.9209166),
    (52.368994, 4.9216975),
    (52.3689584, 4.921786),
    (52.3684874, 4.9231565),
    (52.3681144, 4.924291),
    (52.3689144, 4.9220214),
    (52.3768164, 4.9044924),
    (52.3758586, 4.9076748),
    (52.3758498, 4.9077042),
    (52.375859, 4.9077629),
    (52.3758501, 4.9077777),
    (52.3757993, 4.9085565),
    (52.3765854, 4.9051557),
    (52.3731633, 4.9034895),
    (52.3731365, 4.9035192),
    (52.3716606, 4.9074852),
    (52.3715808, 4.9077209),
    (52.37677, 4.9221595),
    (52.376779, 4.9221594),
    (52.3778307, 4.9153489),
    (52.3774764, 4.9121512),
    (52.3779568, 4.9154211),
    (52.3776451, 4.9161293),
    (52.3774592, 4.9123422),
    (52.3774147, 4.9169689),
    (52.3774512, 4.9125627),
    (52.3656725, 4.9113376),
    (52.3655281, 4.9111924),
    (52.3659792, 4.9116279),
    (52.3660243, 4.9116715),
    (52.3660965, 4.9117296),
    (52.3668432, 4.9119418),
    (52.3667732, 4.9123976),
    (52.3669064, 4.911985),
    (52.3669785, 4.9120431),
    (52.367186, 4.9122464),
    (52.3678174, 4.9127976),
    (52.3665249, 4.9109907),
    (52.3662229, 4.9118603),
    (52.3666445, 4.9116648),
    (52.3659107, 4.9124655),
    (52.3657147, 4.910647),
    (52.3673551, 4.909587),
    (52.3756596, 4.9029189),
    (52.3753986, 4.9028335),
    (52.3741754, 4.9048148),
    (52.3710051, 4.9120735),
    (52.3759465, 4.9027396),
    (52.37592, 4.9028573),
    (52.3717063, 4.9076608),
    (52.370444, 4.9110956),
    (52.3703894, 4.9109641),
    (52.3703349, 4.9108179),
    (52.3700556, 4.9106153),
    (52.3683485, 4.9063168),
    (52.3666151, 4.9247177),
    (52.3708152, 4.9117818),
    (52.3794977, 4.9255823),
    (52.3792283, 4.9233234),
    (52.3714679, 4.9222884),
    (52.3688488, 4.9055626),
    (52.3705156, 4.9087895),
    (52.3704973, 4.908731),
    (52.3763959, 4.9253062),
    (52.3759823, 4.9275867),
    (52.3719517, 4.9265123),
    (52.3759049, 4.9079973),
    (52.3734985, 4.9244259),
    (52.3700608, 4.9187058),
    (52.3668263, 4.9258754)
]

center_lat, center_lon = 52.373275, 4.916235

map_with_circle = folium.Map(location=[center_lat, center_lon], zoom_start=13)

for coordinate in point_coordinates:
    folium.Marker(location=coordinate).add_to(map_with_circle)


folium.Circle(location=[center_lat, center_lon], radius=1000, color='blue', fill=True, fill_opacity=0.2).add_to(map_with_circle)


file_path = "C:\\Users\\eterezakii\\Desktop\\AMS Stuff\\centrality_map_with_circle.html"
map_with_circle.save(file_path)


As with previous questions, we again create a file that can be opened with a web browser. The file contains a map of Amsterdam that has all the horeca points on it.